### PR TITLE
📦 NEW: Login page auth 

### PR DESCRIPTION
**Pull request by waldorfio-dd**
# Description

- ['Swizzled' ](https://refine.dev/docs/packages/documentation/cli/#do-i-need-to-swizzle)the refine/mui Authentication components, to generate editable react components thereby allowing further customisation. Previously, auth was handled by refine/mui's hardcoded login page, which was locked behind their package. Swizzling allows us to customise these components. Would recommend reading the link above as we may swizzle more hardcoded components in the future.
- Replaced default email sign in with Google Oauth sign in (organisation only)
- Adjusted styling of login page to contain DD branding

### Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 📦 New feature (non-breaking change which adds functionality)
- [ ] 👌 Improvement (non-breaking change which improves existing functionality)
- [ ] 🔥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📖 Documentation (non-breaking change which adds documentation)
- [ ] 📖 This change requires a documentation update in Confluence

- [ADP-3] [New]

### Checklist

- [x] Code follows the style guidelines of this project.
- [x] I have performed a self-review and testing.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where required.
- [ ] My changes generate no new warnings or errors.

### Deployment notes 💣 [x] 👍  - [ ] 👎

`yarn install`

### Screenshot or Videos 📷

https://github.com/disrupt-digital/admin-panel/assets/129021106/082167bf-386d-4fe4-a0f8-4fe1ad8834f1

[ADP-3]: https://disruptdigital.atlassian.net/browse/ADP-3?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Test/TestComponent.tsx [hunk 1]
```diff
@@ -1,12 +1,26 @@
-import React from 'react'
-import Slider from '@mui/material/Slider'
+import { AuthPage } from '@refinedev/mui'
 
-export function TestComponent () {
+export function LoginPage () {
```
Comment (andrewk-dd) [inline src/components/Test/TestComponent.tsx:3]: Can you also please rename the file?

File: src/components/auth/components/forgotPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import { useForm } from '@refinedev/react-hook-form'
```
Comment (andrewk-dd) [inline src/components/auth/components/forgotPassword.tsx:1]: It seems like the path `src/components/auth/components/forgotPassword.tsx` can be `src/components/auth/ForgotPasswordPage.tsx`

File: src/components/auth/components/forgotPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  const Link = useLink()
+  const { Link: LegacyLink } = useRouterContext()
+
+  const ActiveLink = routerType === 'legacy' ? LegacyLink : Link
+
+  const PageTitle = title === false ? null : (
+    <img className="w-[200px] pb-[20px] self-center" src="./images/logo_dark.svg" />
+  )
+
+  const Content = (
```
Comment (andrewk-dd) [inline src/components/auth/components/forgotPassword.tsx:45]: Should it be its own component?

File: src/components/auth/components/forgotPassword.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+          <Button type="submit" fullWidth variant="contained" sx={{ mt: "24px", backgroundColor: "#28b1e6" }} disabled={isLoading}>
+            {translate('pages.forgotPassword.buttons.submit', 'Send reset instructions')}
+          </Button>
+        </Box>
+      </CardContent>
+    </Card>
+  )
+
+  return (
+    <>
```
Comment (andrewk-dd) [inline src/components/auth/components/forgotPassword.tsx:99]: It seems like this nesting is not required

File: src/components/auth/components/login.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  })
+  const translate = useTranslate()
+  const routerType = useRouterType()
+  const Link = useLink()
+  const { Link: LegacyLink } = useRouterContext()
+
+  const ActiveLink = routerType === 'legacy' ? LegacyLink : Link
+
+  const PageTitle = title === false ? null : (
+    <img className="w-[200px] pb-[20px] self-center" src="./images/logo_dark.svg" />
```
Comment (andrewk-dd) [inline src/components/auth/components/login.tsx:73]:
```suggestion
    <img className="w-[200px] pb-[20px] self-center" src="/images/logo_dark.svg" />
```

File: src/components/auth/components/login.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  )
+  
+  console.log('providers', providers)
```
Comment (andrewk-dd) [inline src/components/auth/components/login.tsx:76]: 😱

File: src/components/auth/components/login.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+  const renderProviders = () => {
+    if (providers && providers.length > 0) {
```
Comment (andrewk-dd) [inline src/components/auth/components/login.tsx:79]:
```suggestion
    if (!providers || providers.length === 0) {
      return null
    }
```

File: src/components/auth/components/login.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+          >
+            {translate('pages.login.divider', 'or')}
+          </Divider>
+        </>
+      )
+    }
+    return null
+  }
+
+  const Content = (
```
Comment (andrewk-dd) [inline src/components/auth/components/login.tsx:114]: Should it be its own component?

File: src/components/auth/components/register.tsx [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+  const { Link: LegacyLink } = useRouterContext()
+
+  const ActiveLink = routerType === 'legacy' ? LegacyLink : Link
+
+  const PageTitle = title === false ? null : (
+    <img className="w-[200px] pb-[20px] self-center" src="./images/logo_dark.svg" />
+  )
+
+  const renderProviders = () => {
+    if (providers && providers.length > 0) {
```
Comment (andrewk-dd) [inline src/components/auth/components/register.tsx:56]:
```suggestion
    if (!providers || providers.length === 0) {
      return null
    }
```
Something to read: https://dev.to/naucode/the-secret-to-clean-and-efficient-code-how-to-denest-your-code-af1

File: src/components/auth/components/register.tsx [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+            })}
+          </Stack>
+          <Divider sx={{ fontSize: 12, marginY: '16px' }}>{translate('pages.login.divider', 'or')}</Divider>
+        </>
+      )
+    }
+    return null
+  }
+
+  const Content = (
```
Comment (andrewk-dd) [inline src/components/auth/components/register.tsx:90]: Its own component?

File: src/components/auth/components/register.tsx [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+          renderContent(Content, PageTitle)
+        ) : (
+          <>
+            {PageTitle}
+            {Content}
+          </>
+        )}
+      </Container>
+    </Box>
+  )
```
Comment (andrewk-dd) [inline src/components/auth/components/register.tsx:200]: I see it everywhere. Is it worth creating `AuthLayout` or something like that?

File: src/components/auth/index.tsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  onSubmit?: (values: RegisterFormTypes) => void
+}
+
+export type AuthProps = AuthPageProps<BoxProps, CardProps, FormPropsType>
+
+/**
+ * **refine** has a default auth page form served on the `/login` route when the `authProvider` configuration is provided.
+ * @see {@link https://refine.dev/docs/api-reference/mui/components/mui-auth-page/} for more details.
+ */
+export const AuthPage: React.FC<AuthProps> = (props) => {
```
Comment (andrewk-dd) [inline src/components/auth/index.tsx:18]: Should it be responsibility of react router?

**Review by andrewk-dd — APPROVED** [review]
I'm not going to block anything yet as it's under development and "experiment" like

File: src/components/Test/TestComponent.tsx (hunk 1 above)
Comment (waldorfio-dd) [reply src/components/Test/TestComponent.tsx:3]: deleted this component as its no longer required

File: src/components/auth/components/register.tsx (hunk 9 above)
Comment (waldorfio-dd) [reply src/components/auth/components/register.tsx]: thanks, addressed this

File: src/components/auth/components/register.tsx (hunk 11 above)
Comment (waldorfio-dd) [reply src/components/auth/components/register.tsx]: good suggestion, created

File: .eslintrc [hunk 13]
```diff
@@ -45,7 +45,7 @@
       "warn",
       {
         "extensions": [
-          ".tsx"
+          ".tsx", ".jsx"
```
Comment (andrewk-dd) [inline .eslintrc:48]:
```suggestion
          ".tsx",
          ".jsx"
```

**Pull request merged**